mmengine/hooks/checkpoint_hook.py

```diff
@@ -286,6 +294,36 @@ def after_train_epoch(self, runner) -> None:
                 f'Saving checkpoint at {runner.epoch + 1} epochs')
             self._save_checkpoint(runner)
 
+    def after_train_iter(self,
+                         runner,
+                         batch_idx: int,
+                         data_batch: DATA_BATCH = None,
+                         outputs=Optional[dict]) -> None:
+        """Save the checkpoint and synchronize buffers after each iteration.
+
+        Args:
+            runner (Runner): The runner of the training process.
+            batch_idx (int): The index of the current batch in the train loop.
+            data_batch (dict or tuple or list, optional): Data from dataloader.
+            outputs (dict, optional): Outputs from model.
+        """
+
+        if self.by_epoch:
+            return
+
+        if runner.iter + 1 < self.save_begin:
+            return
+
+        # save checkpoint for following cases:
+        # 1. every ``self.interval`` iterations
+        # 2. reach the last iteration of training
+        if self.every_n_train_iters(runner, self.interval) or \
```

Comment:
Why we do not override `every_n_train_iter` anymore? It seems the base `every_n_train_iter` will not consider the `save_begin`